LeetCode/322_coin_change/PJS.cpp

```diff
@@ -0,0 +1,45 @@
+#include <algorithm>
+#include <limits>
+#include <vector>
+using namespace std;
+
+class Solution {
+public:
+    int coinChange(vector<int>& coins, int amount) {
+        if (amount <= 0) {
+            return 0;
+        }
+        
+        if (coins.size() == 1 && amount % coins[0] == 0) {
+            return amount / coins[0];
+        }
+        
+        vector<int> num_min_coins;
+
+        for (int partial_sum = 0 ; partial_sum < amount+1 ; partial_sum++) {
+            num_min_coins.push_back(INT_MAX);
+            for (auto coin : coins) {
+                
+                if (coin == partial_sum) {
+                    num_min_coins[partial_sum] = 1;
+                    continue;
+                }
+                
+                int amount_before_adding = partial_sum - coin;
+                if (amount_before_adding < 0) {
```

Comment:
I like how this variable name is descriptive, but maybe we can shorten it to `amount_before`? Since there is no calculation other than addition?